selfservice/strategy/link/sender.go

```diff
@@ -67,6 +67,12 @@ func (s *Sender) SendRecoveryLink(ctx context.Context, r *http.Request, f *recov
 
 	address, err := s.r.IdentityPool().FindRecoveryAddressByValue(ctx, identity.RecoveryAddressTypeEmail, to)
 	if err != nil {
+		if !s.r.Config().CourierTemplatesRecoveryInvalidSend(ctx) {
```

Comment:
Since this PR was opened, we also added a `code` strategy which is now the default and also needs this feature.